src/modules/user/services/rate.service.ts

```diff
@@ -0,0 +1,33 @@
+import { ForbiddenException, Injectable } from '@nestjs/common';
+import { EErrorMessage } from 'src/common/enums/error-message.enum';
+import { RateRepository } from 'src/modules/repository/services/rate.repository';
+import { UserRepository } from 'src/modules/repository/services/user.repository';
+import { RateRequestDto } from 'src/modules/user/dto/rate.request.dto';
+
+@Injectable()
+export class RateService {
+  constructor(
+    private readonly rateRepository: RateRepository,
+    private readonly userRepository: UserRepository,
+  ) {}
+
+  public async rateUser(rater: string, dto: RateRequestDto): Promise<void> {
```

Comment:
1) look like function name should be updateUserRate 2) and if I understand right uts if user set 5 and then change to 2, I don't think its good option give opportunity to update rate, but even if it happens, I don't think that client should send rate id to backend

Reply:
this method has such name because it is action of rating user by another, not to update existing rate. In our API there is no possibility to change rate, this method has check if current user already rate user with "rate_target_id".

Reply:
but why on line 25 yo uhave update?